app/code/Magento/Sales/view/frontend/layout/sales_order_printinvoice.xml

```diff
@@ -11,6 +11,13 @@
     <update handle="print" />
     <body>
         <attribute name="class" value="account"/>
+        <referenceContainer name="header-wrapper">
+            <referenceBlock name="logo">
+                <arguments>
+                    <argument name="logo_src" xsi:type="helper" helper="Magento\Sales\Helper\Invoice\Logo::getLogoFile"/>
```

Comment:
Please change xsi:type, I believe it should be "object" instead of "helper"
Also for block it's better to use [ViewModel](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/view-models.html)

Reply:
Here I rely on check in original Block class
\Magento\Theme\Block\Html\Header\Logo
    public function getLogoSrc()
    {
        if (empty($this->_data['logo_src'])) {
            $this->_data['logo_src'] = $this->_getLogoUrl();
        }
        return $this->_data['logo_src'];
    }

So I want to inject result of my service class execution into 'logo_src' argument just in one layout. 
I assume, that give me opportunity to change logo for one layout, and not modify original template/block class.